### PR TITLE
feat(ui/conf) : cloud notifications : create listing page 

### PR DESCRIPTION
**Pull request by Med1233**
## Description

**Note for reviewers:** 

This PR is migrated from another module. All feedback in the other PR is made here. for file names, folder architecture and code refactor they are included in the second  PR MON-17198-create-notifications-panel

**Preiview** 

save

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [x] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/packages/ui/src/Dialog/index.tsx [hunk 1]
```diff
@@ -35,6 +35,7 @@ export type Props = {
   confirmDisabled?: boolean;
   contentWidth?: number;
   dialogActionsClassName?: string;
+  dialogConfirmBtnClassName?: string;
```
Comment (bdauria) [inline centreon/packages/ui/src/Dialog/index.tsx:38]:
```suggestion
  dialogConfirmButtonClassName?: string;
```
Avoid shortcuts

File: centreon/packages/ui/src/Form/Inputs/Checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import { ChangeEvent } from 'react';
```
Comment (Thebarda) [inline centreon/packages/ui/src/Form/Inputs/Checkbox.tsx:1]: Please, add some stories for this

File: centreon/packages/ui/src/Form/Inputs/Checkbox.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+}));
+
+const MultiCheckbox = ({
+  change,
+  checkbox,
+  fieldName
+}: InputPropsWithoutGroup): JSX.Element => {
+  const { classes } = useStyles();
+
+  const { values, setFieldValue } = useFormikContext<FormikValues>();
```
Comment (Thebarda) [inline centreon/packages/ui/src/Form/Inputs/Checkbox.tsx:41]: This component must not depend on Formik to work. I would add two props that handle a value and a value setter

File: centreon/packages/ui/src/Form/Inputs/Checkbox.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+  const fieldNamePath = split('.', fieldName);
+
+  const value = path(fieldNamePath, values);
+
+  const handleChange =
+    (label) =>
+    (event: ChangeEvent<HTMLInputElement>): void => {
+      const newValue = value?.map((item) => {
+        if (item.label === label) {
```
Comment (Thebarda) [inline centreon/packages/ui/src/Form/Inputs/Checkbox.tsx:51]:
```suggestion
        if (equals(item.label, label)) {
```

File: centreon/packages/ui/src/Form/Inputs/models.ts [hunk 5]
```diff
@@ -32,6 +35,10 @@ export interface InputProps {
     options: Array<SelectEntry>;
   };
   change?: ({ setFieldValue, value }) => void;
+  checkbox?: {
+    labelPlacement?: LabelPlacement;
```
Comment (Thebarda) [inline centreon/packages/ui/src/Form/Inputs/models.ts:39]: This property is not handled by the Checkbox component

File: centreon/packages/ui/src/Form/Inputs/Checkbox.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            </Box>
+          );
+        })}
+      </FormGroup>
+    ),
+    memoProps: [value]
+  });
+};
+
+export default MultiCheckbox;
```
Comment (Thebarda) [inline centreon/packages/ui/src/Form/Inputs/Checkbox.tsx:96]: Change the file name to MultiCheckbox

File: centreon/packages/ui/src/Form/Inputs/Checkbox.tsx (hunk 2 above)
Comment (Med1233) [reply centreon/packages/ui/src/Form/Inputs/Checkbox.tsx]: I moved the code of this component to another PR, this current PR is only for the Listing component

File: centreon/config/features.json [hunk 7]
```diff
@@ -1,2 +1,3 @@
 {
+    "notification" : 3
```
Comment (Thebarda) [inline centreon/config/features.json:2]:
```suggestion
    "notification" : 0
```

File: centreon/www/front_src/src/CloudNotificationsConfiguration/Actions/DeleteAction.tsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const selected = useAtomValue(selectedRowsAtom);
+
+  const onDeleteActionClick = (): void => setDialogOpen(true);
+
+  const onDeleteActionCancel = (): void => {
+    setDialogOpen(false);
+  };
+
+  const onDeleteActionConfirm = (): void => {
```
Comment (Thebarda) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/Actions/DeleteAction.tsx:37]:
```suggestion
  const closeDeleteActionConfirm = (): void => {
```

File: centreon/www/front_src/src/CloudNotificationsConfiguration/Actions/DeleteAction.tsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+const DeleteAction = (): JSX.Element => {
+  const { classes } = useStyle();
+
+  const { t } = useTranslation();
+
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const selected = useAtomValue(selectedRowsAtom);
+
+  const onDeleteActionClick = (): void => setDialogOpen(true);
```
Comment (Thebarda) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/Actions/DeleteAction.tsx:31]:
```suggestion
  const openDeleteActionConfirm = (): void => setDialogOpen(true);
```

File: centreon/www/front_src/src/CloudNotificationsConfiguration/Actions/DuplicateAction.tsx [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [selected] = useAtom(selectedRowsAtom);
+
+  const onDuplicateActionClick = (): void => setDialogOpen(true);
+
+  const onDuplicateActionCancel = (): void => {
+    setDialogOpen(false);
+  };
+
+  const onDuplicateActionConfirm = (): void => {
```
Comment (Thebarda) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/Actions/DuplicateAction.tsx:37]: Rename function like the previous comments

File: centreon/www/front_src/src/CloudNotificationsConfiguration/EditPanel/index.tsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    height: `calc(100vh - ${theme.spacing(20)})`
+  }
+}));
+
+const EditPanel = (): JSX.Element => {
+  const { classes } = useStyle();
+  const [panelWidth, setPanelWidth] = useAtom(panelWidthStorageAtom);
+  const setIsPanelOpen = useSetAtom(isPanelOpenAtom);
+
+  const handleClose = (): void => setIsPanelOpen(false);
```
Comment (Thebarda) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/EditPanel/index.tsx:25]:
```suggestion
  const close = (): void => setIsPanelOpen(false);
```
Function's name must start with an action verb

File: centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/Actions/ActivateAction.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import { useState } from 'react';
+
+import { Switch as MUISwitch } from '@mui/material';
+
+import type { ComponentColumnProps } from '@centreon/ui';
+
+const ActivateAction = ({ row }: ComponentColumnProps): JSX.Element => {
+  const [checked, setchecked] = useState(row?.isActivated);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
```
Comment (Thebarda) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/Actions/ActivateAction.tsx:10]:
```suggestion
  const change = (event: React.ChangeEvent<HTMLInputElement>): void => {
```

File: centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/Actions/DeleteAction.tsx [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+
+  const [dialogOpen, setDialogOpen] = useState(false);
+
+  const onClick = (): void => setDialogOpen(true);
+
+  const onCancel = (): void => {
+    setDialogOpen(false);
+  };
+
+  const onConfirm = (): void => {
```
Comment (Thebarda) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/Actions/DeleteAction.tsx:34]: Same comment on function name

**Comment by Thebarda** [comment]
The feature branch should be targeted rather than develop

File: centreon/www/front_src/src/CloudNotificationsConfiguration/translatedLabels.ts [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+export const labelActions = 'Actions';
+export const labelStatus = 'Status';
+export const labelSearch = 'Search';
+
+export const labelSelectResourcesAndEvents = 'Select resources and events';
+export const labelSelectUsers = 'Select users';
+export const labelSelectTimePeriodChannelsAndPreview =
+  'Select time period/channels of notifications/preview';
+export const labelDuplicateNotification =
+  'How many times would you like to duplicate selected Notification?';
```
Comment (cg-tw) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/translatedLabels.ts:24]:
```suggestion
  'How many times would you like to duplicate the selected Notification?';
```

File: centreon/www/front_src/src/Resources/Actions/index.test.tsx [hunk 15]
```diff
@@ -523,7 +523,7 @@ describe(Actions, () => {
     ['end', labelEndTime]
   ];
 
-  it.skip.each(invalidDateTestCases)(
+  it.each(invalidDateTestCases)(
     'cannot send a downtime request when the Downtime action is clicked and the %p time input have an invalid format',
```
Comment (cg-tw) [inline centreon/www/front_src/src/Resources/Actions/index.test.tsx]:
```suggestion
    'cannot send a downtime request when the Downtime action is clicked and the %p time input has an invalid format',
```

**Comment by Med1233** [comment]
> Don't forget to add translations in PO files

I have added all translations in panel PR. which includes listing labels

File: centreon/www/front_src/src/CloudNotificationsConfiguration/translatedLabels.ts [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+export const labelActions = 'Actions';
+export const labelStatus = 'Status';
+export const labelSearch = 'Search';
+
+export const labelSelectResourcesAndEvents = 'Select resources and events';
+export const labelSelectUsers = 'Select users';
+export const labelSelectTimePeriodChannelsAndPreview =
+  'Select time period/channels of notifications/preview';
+export const labelDuplicateNotification =
+  'How many times would you like to duplicate the selected Notification?';
```
Comment (cg-tw) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/translatedLabels.ts:24]:
```suggestion
  'How many times would you like to duplicate the selected notification?';
```

File: centreon/www/front_src/src/CloudNotificationsConfiguration/translatedLabels.ts [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+export const labelDeleteNotification = 'Delete Notification?';
```
Comment (cg-tw) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/translatedLabels.ts:25]:
```suggestion
export const labelDeleteNotification = 'Delete notification?';
```

**Review by Thebarda — APPROVED** [review]

**Comment by Med1233** [comment]
hi, @Thebarda can you approve the chromatic UI review please? I think the difference in the wizard button is normal because we are using a story that uses a setTimeout so the button state will change just after loading

**Review by Wedierre — CHANGES_REQUESTED** [review]
Change the color of the trash cans to primary blue and switch to red when hovering.

When the switch to "ON" why is the round green?
Is it like that in MUI?
Capture d’écran 2023-05-30 à 14 28 28

**Comment by Thebarda** [comment]
> hi, @Thebarda can you approve the chromatic UI review please? I think the difference in the wizard button is normal because we are using a story that uses a setTimeout so the button state will change just after loading

Can you apply commands from this link please: https://www.chromatic.com/review?appId=645b4a23e57ed4460d040666&number=889&type=linked&view=changes ?

**Comment by Thebarda** [comment]
> Change the color of the trash cans to primary blue and switch to red when hovering.
> 
> When the switch to "ON" why is the round green? Is it like that in MUI? Capture d’écran 2023-05-30 à 14 28 28

We should probably turn the switch to a primary color

**Comment by Thebarda** [comment]
> hi, @Thebarda can you approve the chromatic UI review please? I think the difference in the wizard button is normal because we are using a story that uses a setTimeout so the button state will change just after loading

In order to avoid confusion on Chromatic, you can juste remove the setTimeout on stories that are impacted

**Review by Wedierre — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Pull request merged**